controllers/reconcile_reverse_proxy.go

```diff
@@ -215,38 +205,288 @@ func (r *ReverseProxyReconciliation) compareEnvVars(existing, desired []v1.EnvVa
+func (r *ReverseProxyReconciliation) applyContainerChanges(current *v1.Container, changes []ContainerChange) bool {
+	needsRestart := false
+
+	for _, change := range changes {
+		r.Log.Info("Applying container change", "field", change.Field, "reason", change.Reason)
+
+		switch change.Field {
+		case "image":
+			current.Image = r.FrontendEnvironment.Spec.ReverseProxyImage
+			needsRestart = true
+		case "env":
+			current.Env = r.getExpectedEnvVars()
+			needsRestart = true
+		case "ports":
+			current.Ports = r.getExpectedPorts()
+			needsRestart = true
+		case "resources":
+			current.Resources = r.getExpectedResources()
+			// Resource changes don't require restart
+		case "volumeMounts":
+			current.VolumeMounts = r.getExpectedVolumeMounts()
+			needsRestart = true
+		case "livenessProbe":
+			current.LivenessProbe = r.getExpectedLivenessProbe()
+			// Probe changes don't require restart
+		case "readinessProbe":
+			current.ReadinessProbe = r.getExpectedReadinessProbe()
+			// Probe changes don't require restart
+		}
+
+		if change.NeedsRestart {
+			needsRestart = true
+		}
+	}
+
+	return needsRestart
+}
+
+// compareVolumeMounts compares two volume mount slices for equality
+func (r *ReverseProxyReconciliation) compareVolumeMounts(current, desired []v1.VolumeMount) bool {
+	if len(current) != len(desired) {
+		return false
+	}
+
+	currentMap := make(map[string]v1.VolumeMount)
+	for _, mount := range current {
+		currentMap[mount.Name] = mount
+	}
+
+	for _, desiredMount := range desired {
+		currentMount, exists := currentMap[desiredMount.Name]
+		if !exists ||
+			currentMount.MountPath != desiredMount.MountPath ||
+			currentMount.ReadOnly != desiredMount.ReadOnly {
+			return false
+		}
+	}
+
+	return true
+}
+
+// Helper methods to get expected values without creating full container
+
+// getExpectedEnvVars returns the expected environment variables
+func (r *ReverseProxyReconciliation) getExpectedEnvVars() []v1.EnvVar {
+	// Get object store configuration
+	objectStoreInfo, err := ExtractBucketConfigFromEnv()
+	if err != nil {
+		// Return empty slice if we can't get config, comparison will detect the difference
+		return []v1.EnvVar{}
+	}
+
+	minioPort := *objectStoreInfo.Port
+	minioEndpoint := *objectStoreInfo.Endpoint
+	bucketPathPrefix := *objectStoreInfo.Name
+	var protocol string
+	switch minioPort {
+	case "443":
+		protocol = "https://"
+	default:
+		protocol = "http://"
+	}
+	minioUpstreamURL := protocol + minioEndpoint + ":" + minioPort
+
+	logLevel := r.FrontendEnvironment.Spec.ReverseProxyLogLevel
+	if logLevel == "" {
+		logLevel = "DEBUG"
+	}
+
+	spaEntrypointPath := r.FrontendEnvironment.Spec.ReverseProxySPAEntrypointPath
+	if spaEntrypointPath == "" {
+		spaEntrypointPath = "/index.html"
+	}
+
+	envVars := []v1.EnvVar{
+		{Name: "SERVER_PORT", Value: strconv.Itoa(ReverseProxyPort)},
+		{Name: "MINIO_UPSTREAM_URL", Value: minioUpstreamURL},
```

Comment:
We should probably rename this VAR at some point in the RP. Seems odd to use in prod.